src/Components/Analyzers/src/ComponentParametersShouldBePublicCodeFixProvider.cs

```diff
@@ -64,9 +64,24 @@ private SyntaxNode HandlePropertyDeclaration(PropertyDeclarationSyntax node)
                 return null;
             }
 
-            var publicModifier = node.Modifiers.FirstOrDefault(m => m.IsKind(SyntaxKind.PublicKeyword));
-            node = node.WithModifiers(
-                node.Modifiers.Remove(publicModifier));
+            var newModifiers = node.Modifiers;
+            for (var i = 0; i < node.Modifiers.Count; i++)
```

Comment:
Consider `for var i = node.Modifiers.Count - 1; i >= 0; i--)`. Your iteration has a bug if you use multiple modifiers such as `protected inten`l or `private protected` since it does not decrement the count

Reply:
node.Modifiers is immutable. It doesn't change.